staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal.go

```diff
@@ -59,6 +60,60 @@ func (s *JSONSchemaPropsOrBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+type cborMajorType int
+
+const (
+	// https://www.rfc-editor.org/rfc/rfc8949.html#section-3.1
+	cborUnsignedInteger cborMajorType = 0
+	cborNegativeInteger cborMajorType = 1
+	cborByteString      cborMajorType = 2
+	cborTextString      cborMajorType = 3
+	cborArray           cborMajorType = 4
+	cborMap             cborMajorType = 5
+	cborTag             cborMajorType = 6
+	cborOther           cborMajorType = 7
+)
+
+const (
+	cborFalseVaue = 0xf4
+	cborTrueValue = 0xf5
+	cborNullValue = 0xf6
+)
+
+func cborType(b byte) cborMajorType {
+	return cborMajorType(b >> 5)
+}
+
+func (s JSONSchemaPropsOrBool) MarshalCBOR() ([]byte, error) {
+	if s.Schema != nil {
+		return cbor.Marshal(s.Schema)
+	}
+	return cbor.Marshal(s.Allows)
+}
+
+func (s *JSONSchemaPropsOrBool) UnmarshalCBOR(data []byte) error {
+	switch {
+	case len(data) == 0:
+		return nil
+	case data[0] == cborFalseVaue, data[0] == cborTrueValue:
+		var b bool
+		if err := cbor.Unmarshal(data, &b); err != nil {
+			return err
+		}
```

Comment:
There's no need to call Unmarshal. Since you can assume `data` is well-formed, this case can be split into two cases that directly return false/true accordingly.